### ISSUE TITLE
Define es_type for PassThroughJSON fields

### ISSUE THREAD

**Issue by amarjandu** [DataBiosphere/azul#2620]
Currently there is no `unmapped_type` (es_type) property defined for PassThroughJSON fields. When attempting to sort on these nested objects there is no clear way for the es_client to know what sort field should be used. Currently requests to sort by these facets raise 500 error responses if the index is empty.

**Comment by amarjandu** [comment]
The solution might require moving to static mappings of facets within ES, this would allow us to clearly define which fields to sort on based on the contents of the facet.

It might be worthwhile to consider a solution that allows inspection of a static mapping for a given facet and expose filters that are applicable to the mapping, it appears that this might be helpful with #2254.

**Comment by amarjandu** [comment]
An in-term solution can be exposing the `document_service` from the `indexer` to the `service.app` to create a list of facets that are able to be sorted (non-nested facets), then use this set as the available sort facets within the API specification.

**Comment by hannes-ucsc** [comment]
> Currently requests to sort by these facets raise 500 error responses if the index is empty.

Do you have an example?

**Comment by amarjandu** [comment]
> > Currently requests to sort by these facets raise 500 error responses if the index is empty.
> 
> Do you have an example?

It can be seen here https://travis-ci.com/github/DataBiosphere/azul/jobs/441223923#L3681 with the addition of the test related to #2396  as part of the `test-then-fix` commit process, because we want to define the `es_type` of PassThroughJSON as None (from this thread: https://github.com/DataBiosphere/azul/pull/2396#discussion_r538094511 ) these facets will still raise an error, and are skipped during the testing process.

Sort by `organismAgeRange`
```
2020-12-11 13:26:41,870 INFO    Thread-35: Received GET request to '/index/files' with parameters {"sort": "organismAgeRange"}.
2020-12-11 13:26:41,880 ERROR   Thread-35: Caught exception for <function get_data at 0x10d6f7790>
Traceback (most recent call last):
  File "/Users/amar/dev/hca/azul/.venv/lib/python3.8/site-packages/chalice/app.py", line 1135, in _get_view_function_response
    response = view_function(**function_args)
  File "/Users/amar/dev/hca/azul/lambdas/service/app.py", line 1153, in get_data
    return repository_search('files', file_id)
  File "/Users/amar/dev/hca/azul/lambdas/service/app.py", line 914, in repository_search
    return service.get_data(catalog=catalog,
  File "/Users/amar/dev/hca/azul/src/azul/service/index_query_service.py", line 66, in get_data
    response = self.transform_request(catalog=catalog,
  File "/Users/amar/dev/hca/azul/src/azul/service/elasticsearch_service.py", line 607, in transform_request
    es_response = es_search.execute(ignore_cache=True)
  File "/Users/amar/dev/hca/azul/.venv/lib/python3.8/site-packages/elasticsearch_dsl/search.py", line 702, in execute
    es.search(
  File "/Users/amar/dev/hca/azul/.venv/lib/python3.8/site-packages/elasticsearch/client/utils.py", line 84, in _wrapped
    return func(*args, params=params, **kwargs)
  File "/Users/amar/dev/hca/azul/.venv/lib/python3.8/site-packages/elasticsearch/client/__init__.py", line 851, in search
    return self.transport.perform_request(
  File "/Users/amar/dev/hca/azul/.venv/lib/python3.8/site-packages/elasticsearch/transport.py", line 351, in perform_request
    status, headers_response, data = connection.perform_request(
  File "/Users/amar/dev/hca/azul/.venv/lib/python3.8/site-packages/elasticsearch/connection/http_urllib3.py", line 231, in perform_request
    self._raise_error(response.status, raw_data)
  File "/Users/amar/dev/hca/azul/.venv/lib/python3.8/site-packages/elasticsearch/connection/base.py", line 229, in _raise_error
    raise HTTP_EXCEPTIONS.get(status_code, TransportError)(
elasticsearch.exceptions.RequestError: RequestError(400, 'x_content_parse_exception', "[1:9724] [field_sort] unmapped_type doesn't support values of type: VALUE_NULL")
2020-12-11 13:26:41,881 DEBUG   Thread-35: Returning 500 response with headers {"Content-Type": "text/plain"}. See next line for the first 1024 characters of the body.
Traceback (most recent call last):
  File "/Users/amar/dev/hca/azul/.venv/lib/python3.8/site-packages/chalice/app.py", line 1135, in _get_view_function_response
    response = view_function(**function_args)
  File "/Users/amar/dev/hca/azul/lambdas/service/app.py", line 1153, in get_data
    return repository_search('files', file_id)
  File "/Users/amar/dev/hca/azul/lambdas/service/app.py", line 914, in repository_search
    return service.get_data(catalog=catalog,
  File "/Users/amar/dev/hca/azul/src/azul/service/index_query_service.py", line 66, in get_data
    response = self.transform_request(catalog=catalog,
  File "/Users/amar/dev/hca/azul/src/azul/service/elasticsearch_service.py", line 607, in transform_request
    es_response = es_search.execute(ignore_cache=True)
  File "/Users/amar/dev/hca/azul/.venv/lib/python3.8/site-packages/elasticsearch_dsl/search.py", line 702, in execute
    es.search(
  File "/Users/amar/dev/hca/azul/.venv/lib/python3.8/site-packages/elasticsearch/client/utils.py", line 84, 
127.0.0.1 - - [11/Dec/2020 13:26:41] "GET /index/files?sort=organismAgeRange HTTP/1.1" 500 -
```

sort by `assayType`
```
127.0.0.1 - - [11/Dec/2020 13:26:41] "GET /index/files?sort=arrayExpressAccessions HTTP/1.1" 200 -
2020-12-11 13:26:41,033 INFO    Thread-4: Received GET request to '/index/files' with parameters {"sort": "assayType"}.
2020-12-11 13:26:41,045 ERROR   Thread-4: Caught exception for <function get_data at 0x10d6f7790>
Traceback (most recent call last):
  File "/Users/amar/dev/hca/azul/.venv/lib/python3.8/site-packages/chalice/app.py", line 1135, in _get_view_function_response
    response = view_function(**function_args)
  File "/Users/amar/dev/hca/azul/lambdas/service/app.py", line 1153, in get_data
    return repository_search('files', file_id)
  File "/Users/amar/dev/hca/azul/lambdas/service/app.py", line 914, in repository_search
    return service.get_data(catalog=catalog,
  File "/Users/amar/dev/hca/azul/src/azul/service/index_query_service.py", line 66, in get_data
    response = self.transform_request(catalog=catalog,
  File "/Users/amar/dev/hca/azul/src/azul/service/elasticsearch_service.py", line 607, in transform_request
    es_response = es_search.execute(ignore_cache=True)
  File "/Users/amar/dev/hca/azul/.venv/lib/python3.8/site-packages/elasticsearch_dsl/search.py", line 702, in execute
    es.search(
  File "/Users/amar/dev/hca/azul/.venv/lib/python3.8/site-packages/elasticsearch/client/utils.py", line 84, in _wrapped
    return func(*args, params=params, **kwargs)
  File "/Users/amar/dev/hca/azul/.venv/lib/python3.8/site-packages/elasticsearch/client/__init__.py", line 851, in search
    return self.transport.perform_request(
  File "/Users/amar/dev/hca/azul/.venv/lib/python3.8/site-packages/elasticsearch/transport.py", line 351, in perform_request
    status, headers_response, data = connection.perform_request(
  File "/Users/amar/dev/hca/azul/.venv/lib/python3.8/site-packages/elasticsearch/connection/http_urllib3.py", line 231, in perform_request
    self._raise_error(response.status, raw_data)
  File "/Users/amar/dev/hca/azul/.venv/lib/python3.8/site-packages/elasticsearch/connection/base.py", line 229, in _raise_error
    raise HTTP_EXCEPTIONS.get(status_code, TransportError)(
elasticsearch.exceptions.RequestError: RequestError(400, 'x_content_parse_exception', "[1:9727] [field_sort] unmapped_type doesn't support values of type: VALUE_NULL")
2020-12-11 13:26:41,048 DEBUG   Thread-4: Returning 500 response with headers {"Content-Type": "text/plain"}. See next line for the first 1024 characters of the body.
Traceback (most recent call last):
  File "/Users/amar/dev/hca/azul/.venv/lib/python3.8/site-packages/chalice/app.py", line 1135, in _get_view_function_response
    response = view_function(**function_args)
  File "/Users/amar/dev/hca/azul/lambdas/service/app.py", line 1153, in get_data
    return repository_search('files', file_id)
  File "/Users/amar/dev/hca/azul/lambdas/service/app.py", line 914, in repository_search
    return service.get_data(catalog=catalog,
  File "/Users/amar/dev/hca/azul/src/azul/service/index_query_service.py", line 66, in get_data
    response = self.transform_request(catalog=catalog,
  File "/Users/amar/dev/hca/azul/src/azul/service/elasticsearch_service.py", line 607, in transform_request
    es_response = es_search.execute(ignore_cache=True)
  File "/Users/amar/dev/hca/azul/.venv/lib/python3.8/site-packages/elasticsearch_dsl/search.py", line 702, in execute
    es.search(
  File "/Users/amar/dev/hca/azul/.venv/lib/python3.8/site-packages/elasticsearch/client/utils.py", line 84, 
127.0.0.1 - - [11/Dec/2020 13:26:41] "GET /index/files?sort=assayType HTTP/1.1" 500 -
```

**Comment by amarjandu** [comment]
Closed in favor of #2621, going to consolidate there.
